diff --git a/node_modules/eslint-mdx/lib/parser.js b/node_modules/eslint-mdx/lib/parser.js
index 1b5bcaf..4fdb4fb 100644
--- a/node_modules/eslint-mdx/lib/parser.js
+++ b/node_modules/eslint-mdx/lib/parser.js
@@ -23,7 +23,11 @@ class Parser {
             ...(0, helpers_1.arrayify)(markdownExtensions),
         ].includes(extname);
         if (!isMdx && !isMarkdown) {
-            throw new Error('Unsupported file extension, make sure setting the `extensions` or `markdownExtensions` option correctly.');
+          return require('@babel/eslint-parser').parseForESLint(code, {
+            filePath,
+            sourceType,
+            requireConfigFile: false,
+          })
         }
         const physicalFilename = (0, helpers_1.getPhysicalFilename)(filePath);
         let result;
